src/crypto/equihash.cpp

```diff
@@ -409,7 +409,11 @@ std::set<std::vector<eh_index>> Equihash<N,K>::OptimisedSolve(const eh_HashState
         // First CollisionByteLength bytes of hash is "stored" as bucket index
         size_t hashLen = N/8 - CollisionByteLength;
         size_t lenIndices = sizeof(eh_trunc);
-        std::array<std::vector<TruncatedStepRow<TruncatedWidth>>, NumBuckets> Xt;
+        std::vector<std::vector<TruncatedStepRow<TruncatedWidth>>> Xt;
+        Xt.reserve(NumBuckets);
```

Comment:
That is correct, and this is probably the main factor in the extended generation time.